File: 0997-find-the-town-judge/0997-find-the-town-judge.java
class Solution {
    public int findJudge(int n, int[][] trust) {
         int dp[]=new int[n+1];
        for (int i=0;i<trust.length;i++){
            dp[trust[i][1]]++;
        }
        int ans=-1;
        for (int i=1;i<=n;i++)
            if(dp[i]==n-1) ans=i;
        if(ans==-1) return -1;
        for (int i = 0; i <trust.length ; i++) {
            if(trust[i][0]==ans) return -1;
        }
        return ans;
    }
}

File: 0997-find-the-town-judge/0997-find-the-town-judge.cpp
class Solution {
public:
    int findJudge(int n, vector<vector<int>>& trust) {
            vector<int>dp(n+1,0);
        for (int i=0;i<trust.size();i++){
            dp[trust[i][1]]++;
             dp[trust[i][0]]--;
        }
        for (int i=1;i<=n;i++)
            if(dp[i]==n-1) return i;
        return -1;
       
    }
};